### PR TITLE
feat: [DHIS2-9280][DHIS2-9279][DHIS2-9282][DHIS2-9284][DHIS2-12575] TEI custom views

### PR DESCRIPTION
**Pull request by simonadomnisoru**
Load, add, update and delete a custom TEI working lists view.

While [DHIS2-12376](https://jira.dhis2.org/browse/DHIS2-12376) and [DHIS2-12574](https://jira.dhis2.org/browse/DHIS2-12574) are being developed, you can test the missing fields by modifying the code locally. For example: 
```
const attributeValueFiltersTest = [
    { attribute: 'coaSpbzZiTB', like: 'abc' },
    { attribute: 'qDQUHqdAXkT', le: '20', ge: '10' },
    { attribute: 'KrCahWFMYYz', in: ['RURAL', 'URBAN'] },
    { attribute: 'TgSJNUL2cqd', in: ['1-2KM'] },
    { attribute: 'mTg1B4t9Liz', in: ['true'] },
    { attribute: 'Z1rLc1rVHK8', in: ['true'] },
];
const enrollmentCreatedDatetest = {
    "startDate": "2014-05-01",
    "endDate": "2019-03-20",
    "type": "ABSOLUTE"
}
const incidentDateTest = { 
    "period": "LAST_WEEK",
    "type": "RELATIVE"
 }; 
const orderTest = 'QRg7SZ6VOAV:desc'; 
const displayColumnOrderTest = ['inactive', 'coaSpbzZiTB', 'QRg7SZ6VOAV', 'Kv4fmHVAzwX']; 
const assignedUserModeTest = 'PROVIDED'; 
const assignedUsersTest = ['DLjZWMsVsq2']; 
```

### CODE REVIEW

File: src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/helpers/TEIFilters/apiTEIFilterToClientConfigConverter/convertPeriodDateToClient.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return PERIODS.LAST_MONTH;
+    }
+    if (
+        periodFrom === moment().subtract(3, 'months').startOf('month').diff(today, 'days') &&
+        periodTo === moment().subtract(1, 'months').endOf('month').diff(today, 'days')
+    ) {
+        return PERIODS.LAST_3_MONTHS;
+    }
+    return null;
+};
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/helpers/TEIFilters/apiTEIFilterToClientConfigConverter/convertPeriodDateToClient.js:40]: Hey, 
According the the [docs](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/tracker.html) `enrollmentCreatedPeriod` and `incidentDate` have a different format then the `dateFilters`. That is why I created a new convertor logic for them.
Thanks!

[the remark this answers is not in the document]
Comment (Bekkalizer) [reply src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/helpers/TEIFilters/apiTEIFilterToClientConfigConverter/convertPeriodDateToClient.js:40]: Hey, enrollmentCreatedPeriod has existed for a while. Any older lists already stored would have the dates stored on this old simpler format. This makes it hard to just change the format on the server side. 
At the same time, the dateFilter format is a more advanced standard better suiting the frontend. If I got this one correctly, the relative period selected here will be translated into a fixed amount of days - which would be correct at the time of storing/reading the filter, but might not be after some days? So a { "periodFrom": -21, "periodTo": 7 } would be "THIS MONTH", but only today. If the same filter is applied tomorrow, would it still work?

File: src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/Setup/TeiWorkingListsSetup.component.js [hunk 2]
```diff
@@ -172,12 +182,18 @@ export const TeiWorkingListsSetup = ({
...
+    onDeleteTemplate,
     ...passOnProps
 }: Props) => {
     const defaultColumns = useDefaultColumnConfig(program, orgUnitId);
     const columns = useColumns<TeiWorkingListsColumnConfigs>(customColumnOrder, defaultColumns);
     const filtersOnly = useFiltersOnly(program);
-    const templates = useStaticTemplates();
+    const apiTemplates = useApiTemplate();
+    const staticTemplates = useStaticTemplates();
+    const templates = apiTemplates?.length > DEFAULT_TEMPLATES_LENGTH ? apiTemplates : staticTemplates;
```
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/Setup/TeiWorkingListsSetup.component.js:189]: Moved to separate issue after discussing: https://jira.dhis2.org/browse/TECH-981

File: src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/helpers/TEIFilters/apiTEIFilterToClientConfigConverter/convertPeriodDateToClient.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/helpers/TEIFilters/apiTEIFilterToClientConfigConverter/convertPeriodDateToClient.js]: Settled in meeting. We will make backend changes and treat `enrollmentCreatedPeriod` as a normal date filter.

**Review by JoakimSM — APPROVED** [review]
LGTM 🎉

**Pull request merged**